### PR TITLE
[ShapeInference] DeformablePSROIPooling shape infer 

### PR DESCRIPTION
**Pull request by mitruska**
### Details:
 - Add shape infer function 
 - Register DeformablePSROIPooling shape infer for CPU
 - Add setters for shape related attributes
 - Add tests

### Tickets:
 - 97217

### CODE REVIEW

File: src/core/shape_inference/include/deformable_psroi_pooling_shape_inference.hpp [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+#pragma once
+#include "openvino/op/deformable_psroi_pooling.hpp"
+
+namespace ov {
+namespace op {
+namespace v1 {
+
+template <class TShape>
+std::vector<TShape> shape_infer(const DeformablePSROIPooling* op, const std::vector<TShape>& input_shapes) {
+    NODE_VALIDATION_CHECK(op, input_shapes.size() >= 2);
```
Comment (praasz) [inline src/core/shape_inference/include/deformable_psroi_pooling_shape_inference.hpp:13]: Change to check size is 2 or 3 as more than 3 is also not correct.

File: src/core/include/openvino/op/deformable_psroi_pooling.hpp [hunk 2]
```diff
@@ -73,9 +73,15 @@ class OPENVINO_API DeformablePSROIPooling : public Op {
     int64_t get_output_dim() const {
         return m_output_dim;
     }
+    void set_output_dim(int64_t output_dim) {
+        m_output_dim = output_dim;
```
Comment (praasz) [inline src/core/include/openvino/op/deformable_psroi_pooling.hpp:77]: Is it fine but adding implementation in cpp is beneficial from compile time perspective as changes in hpp can trigger a lot of files to be rebuild.

**Review by jane-intel — APPROVED** [review]
Please address @praasz comment

File: src/core/shape_inference/include/deformable_psroi_pooling_shape_inference.hpp (hunk 1 above)
Comment (mitruska) [reply src/core/shape_inference/include/deformable_psroi_pooling_shape_inference.hpp]: `>=2` is like a minimum requirement, the algorithm can still works fine for more than 3 inputs, just ignoring the remaining.
Anyway updated to check `2 || 3`.

**Review by praasz — APPROVED** [review]

**Pull request merged**